### PR TITLE
Project setup

### PR DESCRIPTION
**Pull request by tajemouti**
## Hi Reviewer 👋 ##

Here's a summary of the accomplished tasks working on this project:

1. Created a feature branch.
2. Created a separate .rb file for the MyList class.
3. Added a class MyList with an instance variable @list.
4. Implemented the method each in the MyList class.
5. Created a separate .rb file for the MyEnumerable module.
6. Added a module MyEnumerable with the required methods.
7. Created a separate .rb file for test cases.
8. Edited the README and LICENSE files.

### CODE REVIEW

File: my_list.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+class MyList
+  include MyEnumerable
+
+  def initialize(*list)
+    @list = list
+  end
+
+  def each(&block)
+    @list.each(&block)
+  end
```
Comment (ShadyShawkat) [inline my_list.rb]: - [ ] Kindly make sure to implement this method to yield list members from the list. You've done so but using `each`. While this is correct, the purpose of this method is to learn about `yield` and how it works in a method. You can do something like the following to replicate the `each` behavior without actually using it.

```ruby
def each
    return to_enum(:each) unless block_given?

    counter = 0
    while counter < @list.length
      yield(@list[counter])
      counter += 1
    end
    @list
end
```

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi @tajemouti, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: my_list.rb (hunk 1 above)
Comment (tajemouti) [reply my_list.rb]: Thank you @ShadyShawkat.. That was insightful..

**Review by 1GORDON — APPROVED** [review]
# Status: APPROVED 🟢 

Hi @tajemouti, and @Kaiserabbas  ✋

Your project is complete!  There is nothing else to say other than... it's time to merge it: ship it:
Congratulations! :tada: :clap: :clap: :100:

## Highlights!
- MyList class implements #each method that yields all members of @list in succession.✔️

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend considering them as they can improve your code._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**